.generator/templates/model/model.mustache

```diff
@@ -65,11 +67,20 @@ export class {{classname}} {{#parent}}extends {{{parent}}} {{/parent}}{
       if ([{{#enumVars}}{{{value}}}, {{/enumVars}}undefined].includes(data.{{{baseName}}})) {
           res.{{name}} = data.{{{baseName}}};
       } else {
-          throw TypeError(`invalid enum value ${ data.{{{baseName}}} } for {{baseName}}`);
+          console.error(`invalid enum value ${ data.{{{baseName}}} } for {{baseName}}`);
+          throw new ParseException("enum", `invalid enum value ${ data.{{{baseName}}} } for {{baseName}}`);
```

Comment:
I don't think you should raise an error here. An error means you cannot deserialise this object, so you have to put the entire data in `unparsedObject`, and return here (maybe also wipe the attributes you might have already set, if we want to be clean)